### DISCUSSION TITLE
[Platform Request] uBOL for Firefox

### DISCUSSION THREAD

**Discussion by aphilas** [uBlockOrigin/uBOL-home#28] [Q&A] [answered]
I am a "light" uBOL user and I found uBOL on Chrome to be pretty okay for my needs. If I take Google's words at face value, plus a theoretical assumption, the declarative approach should make blocking a lot faster since it happens _inside_ the browser. uBOL [is not in FF Addons Store](https://addons.mozilla.org/en-US/firefox/search/?q=ublock%20lite).

Signing MV3 extensions [should already be available](https://blog.mozilla.org/addons/2022/11/17/manifest-v3-signing-available-november-21-on-firefox-nightly/). I must admit I haven't checked the spec, but I am curious whether there's any blocker preventing packaging uBOL for Firefox.

---

~~I will try packaging uBOL with [`web-ext`](https://github.com/mozilla/web-ext)? and then update this issue.~~

ETA:

Incompatibilities:
- **FF does not yet have support for declarativeNetRequest** - See issue [here](https://bugzilla.mozilla.org/show_bug.cgi?id=1687755) 
- FF only supports [event pages](https://blog.mozilla.org/addons/2022/05/18/manifest-v3-in-firefox-recap-next-steps/) rather than service workers

P.S. I had accidentally hit Enter before adding the body.

**Comment by aphilas** [comment]
@mapx-  Could you please reopen the issue, unless I missed something...

**Comment by mapx-** [comment]
(closed because it was an empty one)

**Comment by mapx-** [comment]
as uBlock Origin Lite only exists for Chrome, I'll convert this "issue" into discussion

**Comment by gorhill** [comment]
Firefox does not have all the APIs yet. I regularly look into this, eventually it will happen.

**Comment by ethpsantos** [comment, marked as answer]
uBOL was released few days ago to Firefox ;)
Thanks devs.
